shinysdr/i/webstatic/client/map/map-layers.js

```diff
@@ -451,17 +459,77 @@ define([
     });
   });
   
-  registerMapPlugin(({addLayer, index}) => {
+  registerMapPlugin(({addLayer, scheduler, index}) => {
+    const smoothStep = 1;
     addLayer('Station Position', {
-      featuresCell: index.implementing('shinysdr.devices.IPositionedDevice'),
-      featureRenderer: function (devicePositioning, dirty) {
-        // TODO: Because devicePositioning is a device component, we don't have the device itself in order to display the device's name. However, the Index is in a position to provide "containing object" information and arguably should.
-        const f = renderTrackFeature(dirty, devicePositioning.track, '');
-        f.iconURL = require.toUrl('./icons/station-user.svg');
-        return f;
-      }
+      featuresCell: new DerivedCell(anyT, scheduler, function(dirty) {
+        let features = [];
+        // TODO: Show only the mapPluginConfig.radio.source device, and not all devices?
+        const devices = index.implementing('shinysdr.devices.IDevice').depend(dirty);
+        devices.forEach(device => {
+          const name = '';  // TODO: Device name doesn't appear to be available?
+          let position;
+          // TODO: There's gotta be a better way to find child components that implement a given interface.
+          const components = device.components.depend(dirty);
+          for (let key in components) {
+            let component = components[key].get();
+            if (isImplementing(component, 'shinysdr.devices.IPositionedDevice')) {
```

Comment:
Instead of using isImplementing, can I rely on the position always being at components.position and the rotator always being at components.rotator? The code I'm replacing here did not, but the deviceTrack function earlier in the file relies on the position being at components.position.